### PR TITLE
New logic2

### PR DESCRIPTION
**Pull request by lukemattle**
should be finished now :)

### CODE REVIEW

**Comment by lukemattle** [comment]
for some reason my save locally messed up slightly - i had a backup so have sorted it locally but let me know if any of this is not as expected

File: frontend/scripts/gameLogicInterface.js [hunk 1]
```diff
@@ -99,23 +101,94 @@ export const emptyGameState = {
...
+			}
+			
 		},
 
 		/**
 		 * Return if the game is over
 		 * @return {boolean}
 		 */
 		isGameOver: function() {
+			if (playfield[19].includes("cyan" || "blue" || "orange" || "yellow" || "green" || "purple" || "red")){
```
Comment (josephcwalker) [inline frontend/scripts/gameLogicInterface.js]: I don't think this syntax is valid. I might be wrong but could you show me where you found this?

I think you might have to re-write this in a different way (you've done it nicely in other places)

File: frontend/scripts/gameLogicInterface.js [hunk 2]
```diff
@@ -155,7 +228,8 @@ export default function createGame(initialGameState = emptyGameState) {
 		 * Which stats to keep track of can be determined later
 		 */
 		getStats: function() {
-			
+			return this.gameState.tetrisesMade
```
Comment (josephcwalker) [inline frontend/scripts/gameLogicInterface.js]: Two returns one after the other won't return both things. It'll only return the first one and then never run the second one.

Comment (lukemattle) [reply frontend/scripts/gameLogicInterface.js]: fixed:
image

File: frontend/scripts/gameLogicInterface.js (hunk 1 above)
Comment (lukemattle) [reply frontend/scripts/gameLogicInterface.js]: fixed

**Review by josephcwalker — APPROVED** [review]

**Pull request merged**